### DISCUSSION TITLE
Name migration files so they appear chronologically

### DISCUSSION THREAD

**Discussion by sondrelg** [sqlalchemy/alembic#961] [Ideas]
Hi,

First of all thanks for creating and maintaining alembic - it's great! 👏 

After getting to to know the tool, I only really have one pain-point during development: **My migration files are ordered completely randomly**, making it much harder to make sense of them when I want to manually browse the different versions.

![image](https://user-images.githubusercontent.com/25310870/140886625-fd3c2966-f1e2-44af-a3f2-69d17f7a1fed.png)

I assume you're just generating an ID for each version, but I was wondering, would it technically possible (and easy) to change the format to one where the versions appear in chronological order? Off the top of my head, maybe just prepending something before the ID, like `{timestamp}_{id_}_{message}` or just `{migration_number}_{id_}_{message}` could make sense?

Definitely understand if there are reasons why this would be impractical, but thought I'd ask, just in case it's not 🙂  Also, if there are better ways of browsing versions/if I'm using the tool incorrectly somehow, please let me know!

**Comment by zzzeek** [comment]
hi there -

yes, this is a supported feature.  look in [alemic.ini](https://alembic.sqlalchemy.org/en/latest/tutorial.html#editing-the-ini-file) and set up `file_template`.  scroll down in that doc section below the example and file_template is described along with how to put datestamps in there.

**Reply by sondrelg** [reply]
How nice! Thanks for the quick reply 👏
